src/nlsolve/newton.jl

```diff
@@ -234,7 +166,150 @@ end
         integrator.destats.nf += 1
     end
 
+    b, ustep = _compute_rhs!(nlsolver, integrator, f, z)
+
+    # update W
+    if W isa DiffEqBase.AbstractDiffEqLinearOperator
+        update_coefficients!(W, ustep, p, tstep)
+    end
+
+    if integrator.opts.adaptive
+        reltol = integrator.opts.reltol
+    else
+        reltol = eps(eltype(dz))
+    end
+
+    if is_always_new(nlsolver) || (iter == 1 && new_W)
+        linres = dolinsolve(integrator, linsolve; A = W, b = _vec(b), linu = _vec(dz),
+                            reltol = reltol)
+    else
+        linres = dolinsolve(integrator, linsolve; A = nothing, b = _vec(b), linu = _vec(dz),
+                            reltol = reltol)
+    end
+
+    cache.linsolve = linres.cache
+
+    if DiffEqBase.has_destats(integrator)
+        integrator.destats.nsolve += 1
+    end
+
+    # relaxed Newton
+    # Diagonally Implicit Runge-Kutta Methods for Ordinary Differential
+    # Equations. A Review, by Christopher A. Kennedy and Mark H. Carpenter
+    # page 54.
     if isdae
+        γdt = α * invγdt
+    else
+        γdt = γ * dt
+    end
```

Comment:
Why is the relaxation only applied to DAEs?

Reply:
It's not, this is pre-existing code. I am not sure what alpha refers to here but it's not the relaxation constant. Should we rename the variable in `relax!` to avoid confusion? @YingboMa